CAMEL-19491: aws2-sqs - use `ScheduledPollConsumerHealthCheck`

simon-ras opened the pull request:
The SQS consumer is a polling consumer, and thus we can use `ScheduledPollConsumerHealthCheck` where we observe the last poll rather than making a new operation against SQS. Previously `listQueues` was called, which is a different API than what the consumer actually uses (`receiveMessage`).

The `Sqs2ConsumerHealthCheck` used to check if configured region existed in the metadata of the client. That logic was copied over from the old `Sqs2ComponentVerifierExtension`. It is removed from this implementation as the call to `poll` will yield in the same outcome in case an incorrect Region is defined in the configuration. In this case, a failed poll, causing the health check to be DOWN.

<!--
- Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
-->

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

<!--
# *Note*: trivial changes like, typos, minor documentation fixes and other small items do not require a JIRA issue. In this case your pull request should address just this issue, without pulling in other changes.
-->

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

<!--
If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
-->

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

<!--
You can run the aforementioned command in your module so that the build auto-formats your code. This will also be verified as part of the checks and your PR may be rejected if if there are uncommited changes after running `mvn clean install -DskipTests`.

You can learn more about the contribution guidelines at https://github.com/apache/camel/blob/main/CONTRIBUTING.md
-->

simon-ras commented on components/camel-aws/camel-aws2-sqs/src/main/java/org/apache/camel/component/aws2/sqs/Sqs2Consumer.java:
```diff
@@ -375,16 +371,6 @@ protected void doStart() throws Exception {
-
-        if (healthCheckRepository != null) {
-            consumerHealthCheck = new Sqs2ConsumerHealthCheck(this, getRouteId());
-            consumerHealthCheck.setEnabled(getEndpoint().getComponent().isHealthCheckEnabled() && getEndpoint().getComponent().isHealthCheckConsumerEnabled());
```
This solution does not honor these configuration options. Should they be removed, or should I override `doBuild()` in order to only add the health check in case it is configured?

davsclaus replied:
Ah these new options are added in v4 - I am moving this to the scheduled poll consumer, so in your PR you can just remove these lines.

davsclaus replied:
https://github.com/apache/camel/pull/10849

simon-ras commented:
Please be aware that there is a few more aws consumers which could benefit from the same solution e.g. AWS2S3Consumer

davsclaus commented:
Yes @simon-ras yes can you update the PR with the other aws components that should have done the same as this, then we can have them all in the same PR

simon-ras commented:
> Yes @simon-ras yes can you update the PR with the other aws components that should have done the same as this, then we can have them all in the same PR

Done, added as separate commits, also updated original jira to explain that other components was affected.

oscerd commented:
Thanks for this, having more people reporting and testing AWS/GCP/Azure components is really important.

davsclaus commented:
org.apache.camel.builder.endpoint.FileConsumeCharsetTest.testConsumeAndDelete | 0.199s | FAILURE
-- | -- | --

davsclaus commented:
test not related
